### ISSUE TITLE
Samsung Galaxy A12 bootloop problems with the lastest Magisk release.

### ISSUE THREAD

**Issue by overlay-fs** [topjohnwu/Magisk#8643]
I decided to root my Samsung Galaxy A12 again without noticing the release of Magisk 28.1, I followed the steps to the letter, and I got an unpleasant surprise, when flashing the magisk patch with odin, and reboot, this disastrous bootloop occurs. Of course I went to recovery first and factory reset. Then I flashed the firmware again and decided to flash the Magisk 28.0 patch, and guess what happened: it worked without problems.

**It can be a serious problem for inexperienced or novice users. At the moment my recommendation is not to upgrade to Magisk 28.1.** _I report the problem for consideration._

Device: Samsung Galaxy A12 (A127M/DS)
Android version: Android 13
Magisk version name: Magisk 28.1
Magisk version code: Magisk 28101

**Comment by LorenzoLX** [comment]
i cant exit from the bootloop

**Comment by harvald** [comment]
I have similar problem. 
Device: Samsung Galaxy M15 (SM-M156B/DS)
Android version: Android 14
Magisk version name: Magisk 28.1
Magisk version code: Magisk 28100

Only in my case I cannot enter recovery, so cannot do factory reset. Only download mode works. Flashing stock firmware did not help.

**Comment by A-Australian** [comment]
> i cant exit from the bootloop

1.Connect to PC
2. When phone turns off, quickly hold power up and down
3. When in download mode, flash stock (all)
4. Quickly go into recovery mode after flashing
5. Factory reset

**Comment by leo-b** [comment]
I also ran into a boot loop with my old Galaxy S10 with the very old G973FXXUFHVE1.
Luckily I found a backup of my old recovery img to flash again.

**Comment by AhmedMagisk** [comment]
did you tried with canary build ?

**Comment by leo-b** [comment]
> did you tried with canary build ?
Yes. The reboot loop occured after an upgrade from 28102 which worked flawlessly to 28103.

**Comment by hamed7ir** [comment]
Same here I used, latest official and then latest beta version of magisk that released 2 weeks ago, both caused boot looping on my s21 
I used latest android 15 files for my device, if you need logs I can send you. 
I noticed that ap file size reduces after patch via magisk about 1 GB.

**Comment by wrlopez** [comment]
I can confirm that in my A12 with magisk 28.0 it works. Thanks @capitulador

**Comment by Yannick10015** [comment]
> I can confirm that in my A12 with magisk 28.0 it works. Thanks @capitulador
> 
> edit: I forgot to mention that I tried multiple times with 29.0 and always got the bootloop

i have a A125FXXS6CXJ5 and tried to root it with v.28.0 but i got always into bootloop do you know a solution maybe or anyone else?

**Comment by ronnieyadaoguillermo06-byte** [comment]
i have a A125FXXS6CXJ5 and tried to root it with v.28.0 but i got always into bootloop do you know a solution maybe or anyone else?

**Comment by DBC201** [comment]
I also had this issue and it was solved by using v28.0.

I tried latest pre-release and then v29 but neither worked. Trying v28.0 finally worked. Tried pre-release and v29 with firmware A127FXXSDDXJ1 (original firmware) and A127FXXSDDXJ5 (my desired firmware) but neither worked. With v28.0 I was able to do it with my desired firmware in a single try.

**Comment by harvald** [comment]
In my case problem was solved with the newest samsung firmware. I had to wait half year with bricked phone!

**Comment by 11philip22** [comment]
> In my case problem was solved with the newest samsung firmware. I had to wait half year with bricked phone!

what firmware version is that and where do I get it. Currently in a bootloop with magisk 28 and the firmware version Frija downloaded :(

**Comment by harvald** [comment]
I downloaded it from samfw.com 
Newest at that time was from 2025-08-01. First I've tried with Android 15.

**Comment by yashyhy01-dot** [comment]
> قررتُ تثبيت روت لهاتفي سامسونج جالاكسي A12 مجددًا دون أن ألاحظ إصدار Magisk 28.1. اتبعتُ الخطوات حرفيًا، لكن مفاجأة غير سارة حدثت عند تثبيت Magisk باستخدام برنامج Odin، وعند إعادة التشغيل، حدثت حلقة إعادة تشغيل كارثية. بالطبع، ذهبتُ إلى وضع الاسترداد أولًا وأعدتُ ضبط المصنع. ثم ثبّتُ البرنامج الثابت مجددًا وقررتُ تثبيت Magisk 28.0، وتخيلوا ما حدث: نجح الأمر دون مشاكل.
> 
> **قد تُشكّل هذه المشكلة مشكلةً خطيرةً للمستخدمين المبتدئين أو عديمي الخبرة. أنصح حاليًا بعدم الترقية إلى Magisk 28.1. ** _أُبلغ عن المشكلة للنظر فيها._
> 
> الجهاز: Samsung Galaxy A12 (A127M/DS) إصدار أندرويد: Android 13 اسم إصدار Magisk: Magisk 28.1 رمز إصدار Magisk: Magisk 28101

**Comment by yashyhy01-dot** [comment]
#8927

**Comment by yashyhy01-dot** [comment]
https://github.com/topjohnwu/Magisk/issues/8643#issue-2745223577

**Comment by aniruddha-ga** [comment]
bootloop problem

Device: Samsung A12
Android Version: Android 12
Model: SM-125F/DS
AP: A125FXXS6CXJ1

Tried Magisk Verisons: 28.0, 28.1, 30.0, 25, and tried latest 30.6
Result: Bootloop

Firmware downloaded from: samfw
Works only without patch Orignal but after patch with Magisk bootloop

Software used to flash: [Odin 3.14.4](https://odindownload.com/SamsungOdin/)

**Comment by francesconovo43-code** [comment]
@aniruddha-ga stessa cosa a me puoi dirmi per favore se sei riuscito alla fine?
